state/protocol/badger/snapshot.go

```diff
@@ -64,15 +69,76 @@ func (s *Snapshot) Identities(selector flow.IdentityFilter) (flow.IdentityList,
 		return nil, err
 	}
 
-	// TODO: CAUTION SHORTCUT
-	// We report the initial identities as of the EpochSetup event here.
-	// Only valid as long as we don't have slashing
-	identities := setup.Participants.Filter(selector)
+	// get identities from the current epoch first
+	identities := setup.Participants
+	lookup := identities.Lookup()
+
+	// get identities that are in either last/next epoch but NOT in the current epoch
+	var otherEpochIdentities flow.IdentityList
+	phase := status.Phase()
+	switch phase {
+	// during staking phase (the beginning of the epoch) we include identities
+	// from the previous epoch that are now un-staking
+	case flow.EpochPhaseStaking:
+
+		first, err := s.state.AtBlockID(status.FirstBlockID).Head()
+		if err != nil {
+			return nil, fmt.Errorf("could not get first block of epoch: %w", err)
+		}
+		// parent ID may zero if the first block in this epoch is the root block
+		// in this case there are no previous epoch identities to check anyway
+		if first.ParentID == flow.ZeroID {
+			break
+		}
```

Comment:
⚠️ 
I don't think this logic would work for our current spork process:
* the root block of a spork has as parent a pre-spork block, which should _not_ be `ZeroID` 

#### suggestion:
```suggestion
		
		// parent ID may zero if the first block in this epoch is the root block
		// in this case there are no previous epoch identities to check anyway
		stateParams := Params{state: s.state}
		root, err := stateParams.Root()
		if err != nil {
			return nil, fmt.Errorf("could not get global state parameters: %w", err)
		}
		if root.Height == first.Height {
			break
		}
```

Reply:
Good catch 🙏 😓

Reply:
https://github.com/onflow/flow-go/pull/143/commits/24c35ff43bf33570643f7de86e450e9ef2596e16

Reply:
thanks for adding a test 🎸